storage/block/block.go

```diff
@@ -176,45 +176,50 @@ func (b *dbBlock) Stream(blocker context.Context) (xio.SegmentReader, error) {
 
 	b.ctx.DependsOn(blocker)
 
+	start := b.startWithLock()
+	end := b.LastReadTime()
```

Comment:
The end time of the block is actually start + block size.

The last read time is when the block was last accessed.

Hm, we might need to make the block constructor take a start time and an end time, along with the Reset(...) call.